src/display/editor/annotation_editor_layer.js

```diff
@@ -679,14 +679,32 @@ class AnnotationEditorLayer {
 
   /**
    * SelectionChange callback.
-   * @param {Event} _event
+   * @param {Event} event
    */
-  selectionStart(_event) {
-    this.#textLayer?.div.addEventListener(
-      "pointerup",
-      this.#boundPointerUpAfterSelection,
-      { once: true }
-    );
+  selectionStart(event) {
+    if (
+      !this.#textLayer ||
+      event.target.parentElement.closest(".textLayer") !== this.#textLayer.div
+    ) {
+      return;
+    }
+
+    if (this.#uiManager.isShiftKeyDown) {
+      const keyup = () => {
+        if (this.#uiManager.isShiftKeyDown) {
+          return;
+        }
+        window.removeEventListener("keyup", keyup);
+        this.pointerUpAfterSelection({});
+      };
+      window.addEventListener("keyup", keyup);
```

Comment:
Given that this is a "global" event listener, can we be guaranteed that it'll *always* be removed as expected?
What about the following, perhaps contrived, situation:
 1. Enable the Highlight-editor.
 2. Press *and hold* the <kbd>Shift</kbd> key.
 3. Click outside of the browser window.
 4. Release the <kbd>Shift</kbd> key.

It seems, unless I'm missing something, that the `keyup` event won't fire and this listener will thus not be removed as expected in that case.